src/token.rs

```diff
@@ -85,26 +131,50 @@ pub mod contract {
 
         /// Returns the current balance for some address.
         fn balanceOf(&mut self, owner: Address) -> U256 {
-            balance_of(&owner)
+            read_balance_of(&owner)
         }
 
         /// Transfer funds
         fn transfer(&mut self, to: Address, amount: U256) -> bool {
             let sender = ext::sender();
-            let mut senderBalance = balance_of(&sender);
-            let mut recipientBalance = balance_of(&to);
+            let senderBalance = read_balance_of(&sender);
+            let recipientBalance = read_balance_of(&to);
             if amount == 0.into() || senderBalance < amount {
                 false
             } else {
-                senderBalance = senderBalance - amount;
-                recipientBalance = recipientBalance + amount;
-                storage::write(&balance_key(&sender), &senderBalance.into());
-                storage::write(&balance_key(&to), &recipientBalance.into());
+                storage::write(&balance_key(&sender), &(senderBalance - amount).into());
+                storage::write(&balance_key(&to), &(recipientBalance + amount).into());
```

Comment:
Would it be better to extract expressions like `recipientBalance + amount` into thier own statements. I personally think that these operations are pretty important to be maximally visible for readers of the code.

Reply:
(here and below)